playwright/src/main/java/com/microsoft/playwright/junit/impl/PlaywrightExtension.java

```diff
@@ -4,18 +4,50 @@
 import com.microsoft.playwright.junit.Options;
 import org.junit.jupiter.api.extension.*;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
 import static com.microsoft.playwright.junit.impl.ExtensionUtils.isParameterSupported;
 import static com.microsoft.playwright.junit.impl.ExtensionUtils.setTestIdAttribute;
 
-public class PlaywrightExtension implements ParameterResolver, AfterAllCallback {
-  private static final ThreadLocal<Playwright> threadLocalPlaywright = new ThreadLocal<>();
+public class PlaywrightExtension implements ParameterResolver, BeforeAllCallback, ExtensionContext.Store.CloseableResource {
+  private static final ThreadLocal<Playwright> threadLocalPlaywright;
+  private static final List<Playwright> playwrightList;
+  private final static Lock beforeLock;
+  private static boolean isTestRunStarted;
+
+  static {
+    isTestRunStarted = false;
+    beforeLock = new ReentrantLock();
+    threadLocalPlaywright = new ThreadLocal<>();
+    playwrightList = new ArrayList<>();
```

Comment:
It should be `Collections.synchronizedList` as there is a chance it is accessed from multiple threads concurrently.

Reply:
makes sense i will update it.

Reply:
done